query/query3_test.go

```diff
@@ -561,6 +562,359 @@ func TestKShortestPathWeighted1MinMaxWeight(t *testing.T) {
 	`, js)
 }
 
+func TestKShortestPathDepthNoPath(t *testing.T) {
+	query := `
+	{
+		A as shortest(from: 1, to:1000, numpaths: 2, depth:2) {
+			follow
+		}
+		me(func: uid(A)) {
+			name
+		}
+	}`
+	js := processQueryNoErr(t, query)
+	require.JSONEq(t, `{"data": {"me":[]}}`, js)
+}
+
+func TestKShortestPathDepthOnePath(t *testing.T) {
+	query := `
+	{
+		A as shortest(from: 1, to:1000, numpaths: 2, depth:3) {
+			follow
+		}
+		me(func: uid(A)) {
+			name
+		}
+	}`
+	js := processQueryNoErr(t, query)
+	require.JSONEq(t, `{
+		"data": {
+		  "me": [
+			{
```

Comment:
can these be compressed a bit to not use so much vertical space
me: [
  {"name": "x"},
  {"name": "y"}
..
]